From db466e232e916de29e54e9ff479de2cbcd278d2f Mon Sep 17 00:00:00 2001
From: Dabo Ross <daboross@daboross.net>
Date: Sun, 13 Apr 2014 16:14:00 -0700
Subject: [PATCH] Move formatting of domain in domains plugin to separate
 function, and fix having a comma appear after the last result


diff --git a/plugins/domainr.py b/plugins/domainr.py
index e853bfa..e90747f 100644
--- a/plugins/domainr.py
+++ b/plugins/domainr.py
@@ -1,18 +1,35 @@
 from util import hook, http
 
+formats = {
+    "taken": "\x034{domain}\x0f{path}",
+    "available": "\x033{domain}\x0f{path}",
+    "other": "\x031{domain}\x0f{path}"
+}
 
-@hook.command
+
+def format_domain(domain):
+    """
+    :type domain: dict[str, str]
+    """
+    if domain["availability"] in formats:
+        domainformat = formats[domain["availability"]]
+    else:
+        domainformat = formats["other"]
+    return domainformat.format(**domain)
+
+
+@hook.command()
+@hook.command("domain")
 def domainr(inp):
-    """domainr <domain> - Use domain.nr's API to search for a domain, and similar domains."""
+    """domainr <domain> - Use domain.nr's API to search for a domain, and similar domains.
+    :type inp: str
+    """
     try:
         data = http.get_json('http://domai.nr/api/json/search?q=' + inp)
     except (http.URLError, http.HTTPError) as e:
         return "Unable to get data for some reason. Try again later."
     if data['query'] == "":
         return "An error occurred: {status} - {message}".format(**data['error'])
-    domains = ""
-    for domain in data['results']:
-        domains += ("\x034" if domain['availability'] == "taken" else (
-                    "\x033" if domain['availability'] == "available" else "\x031")) + domain['domain'] + "\x0f" + domain[
-                    'path'] + ", "
-    return "Domains: " + domains
+
+    domains = [format_domain(domain) for domain in data["results"]]
+    return "Domains: {}".format(", ".join(domains))
-- 
1.9.2

